WordPress/Classes/ViewRelated/Likes/LikesListController.swift

```diff
@@ -0,0 +1,343 @@
+            return errorCell
+        }
+
+        return userCell(for: indexPath)
+    }
+
+    func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        return Constants.estimatedRowHeight
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        if indexPath.section == Constants.headerSectionIndex {
+            delegate?.didSelectHeader()
+            return
+        }
+
+        guard !isLoadingContent,
+              !isShowingError,
+              indexPath.row < likingUsers.count else {
+            return
+        }
+
+        delegate?.didSelectUser(likingUsers[indexPath.row])
+    }
+
+}
+
+// MARK: - Notification Cell Handling
+
+private extension LikesListController {
+
+    func headerCell() -> NoteBlockHeaderTableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: NoteBlockHeaderTableViewCell.reuseIdentifier()) as? NoteBlockHeaderTableViewCell,
+              let group = notification?.headerAndBodyContentGroups[Constants.headerRowIndex] else {
+            DDLogError("Error: couldn't get a header cell or FormattableContentGroup.")
+            return NoteBlockHeaderTableViewCell()
+        }
+
+        setupHeaderCell(cell: cell, group: group)
+        return cell
+    }
+
+    func setupHeaderCell(cell: NoteBlockHeaderTableViewCell, group: FormattableContentGroup) {
+        cell.attributedHeaderTitle = nil
+        cell.attributedHeaderDetails = nil
+
+        guard let gravatarBlock: NotificationTextContent = group.blockOfKind(.image),
+            let snippetBlock: NotificationTextContent = group.blockOfKind(.text) else {
+                return
+        }
+
+        cell.attributedHeaderTitle = formatter.render(content: gravatarBlock, with: HeaderContentStyles())
+        cell.attributedHeaderDetails = formatter.render(content: snippetBlock, with: HeaderDetailsContentStyles())
+
+        // Download the Gravatar
+        let mediaURL = gravatarBlock.media.first?.mediaURL
+        cell.downloadAuthorAvatar(with: mediaURL)
+    }
+
+    func userCell(for indexPath: IndexPath) -> NoteBlockUserTableViewCell {
+        guard indexPath.row < likingUsers.count,
+              let cell = tableView.dequeueReusableCell(withIdentifier: NoteBlockUserTableViewCell.reuseIdentifier()) as? NoteBlockUserTableViewCell else {
+            DDLogError("Error: couldn't get a user cell or requested row is out of boundary")
+            return NoteBlockUserTableViewCell()
+        }
+
+        let user = likingUsers[indexPath.row]
+        cell.accessoryType = .none
+        cell.name = user.displayName
+
+        // TODO: Re-enable follow functionality once the information is available.
```

Comment:
🤔 I don't think we are going to re-enable it, correct? Per the design, the follow button on the cell has been removed, and can be accessed via the user's site. So this comment isn't accurate?

Reply:
Oh, got it. I assumed that we wanted to match the previous look first, instead of building it towards the new cell designs directly. Will fix this in the next commit!

Reply:
Yea, generally that was the idea. But I think going ahead as-is is worth not having to add endpoint support for something that would be short lived. Thank you!